platipy/imaging/label/comparison.py

```diff
@@ -293,3 +293,94 @@ def compute_metric_hd(label_a, label_b, auto_crop=True):
+
+    Returns:
+        float: The total (slice-wise) added path length in mm
+    """
+
+    added_path_length_list = []
+    n_slices = label_ref.GetSize()[2]
+
+    # convert the distance threshold to voxel units
+    distance_threshold = int(np.ceil(distance_threshold_mm / np.mean(label_ref.GetSpacing()[:2])))
+
+    # iterate over each slice
+    for i in range(n_slices):
+
+        if (sitk.GetArrayViewFromImage(label_ref)[i].sum() + sitk.GetArrayViewFromImage(label_ref)[i].sum()) == 0:
+            continue
+
+        label_ref_contour = sitk.LabelContour(label_ref[:,:,i])
+        label_test_contour = sitk.LabelContour(label_test[:,:,i])
+
+        if distance_threshold_mm > 0:
+            label_ref_contour = sitk.BinaryDilate(label_ref_contour, (distance_threshold,distance_threshold))
+
+        # mask out the locations in agreement
+        added_path = sitk.MaskNegated(label_test_contour, label_ref_contour)
+
+        # add up the voxels on the added path
+        added_path_length = sitk.GetArrayViewFromImage(added_path).sum()
+        added_path_length_list.append(added_path_length)
+
+    total_added_path_length = np.sum(added_path_length_list)
+    total_added_path_length_mm = total_added_path_length * np.mean(label_ref.GetSpacing()[:2])
+
+    return total_added_path_length_mm
+
+def compute_metric_mean_apl(label_ref, label_test, distance_threshold_mm=0):
+    """Compute the mean (slice-wise) added path length in mm
+
+    This operates on transaxial slices, which are assumed to be in the z-dimension (axis=2).
+
+    Args:
+        label_ref (sitk.Image): The reference (ground-truth) label
+        label_test (sitk.Image): The test label
+        distance_threshold_mm (float): Distances under this threshold will not contribute to the added path length
+
+    Returns:
+        float: The mean (slice-wise) added path length in mm
+    """
+
+    added_path_length_list = []
+    n_slices = label_ref.GetSize()[2]
+
+    # convert the distance threshold to voxel units
+    distance_threshold = int(np.ceil(distance_threshold_mm / np.mean(label_ref.GetSpacing()[:2])))
+
+    # iterate over each slice
+    for i in range(n_slices):
+
+        if (sitk.GetArrayViewFromImage(label_ref)[i].sum() + sitk.GetArrayViewFromImage(label_ref)[i].sum()) == 0:
+            continue
+
+        label_ref_contour = sitk.LabelContour(label_ref[:,:,i])
+        label_test_contour = sitk.LabelContour(label_test[:,:,i])
+
+        if distance_threshold_mm > 0:
+            label_ref_contour = sitk.BinaryDilate(label_ref_contour, (distance_threshold,distance_threshold))
+
+        # mask out the locations in agreement
+        added_path = sitk.MaskNegated(label_test_contour, label_ref_contour)
+
+        # add up the voxels on the added path
+        added_path_length = sitk.GetArrayViewFromImage(added_path).sum()
+        added_path_length_list.append(added_path_length)
+
+    mean_added_path_length = np.mean(added_path_length_list)
```

Comment:
Are these two functions identical expect for this line? Could be combined into one function? Or at least pull the common bits out into one function they both call?

Reply:
yeah they are, let's definitely do that!

Reply:
Hey Phil,

Great comments, thanks! I've refactored the code, and also set the default parameter to 3mm.